crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -141,38 +145,198 @@ impl Persistence {
     pub async fn auction_prices(
         &self,
         auction: domain::auction::Id,
-    ) -> Result<HashMap<eth::TokenAddress, domain::auction::Price>, Error> {
+    ) -> Result<HashMap<eth::TokenAddress, domain::auction::Price>, error::Auction> {
         let mut ex = self
             .postgres
             .pool
             .begin()
             .await
             .context("begin")
-            .map_err(Error::DbError)?;
+            .map_err(error::Auction::DbError)?;
 
         let db_prices = database::auction_prices::fetch(&mut ex, auction)
             .await
             .context("fetch")
-            .map_err(Error::DbError)?;
+            .map_err(error::Auction::DbError)?;
 
         let mut prices = HashMap::new();
         for price in db_prices {
             let token = eth::H160(price.token.0).into();
             let price = big_decimal_to_u256(&price.price)
                 .ok_or(domain::auction::InvalidPrice)
                 .and_then(|p| domain::auction::Price::new(p.into()))
-                .map_err(Error::Price)?;
+                .map_err(error::Auction::Price)?;
             prices.insert(token, price);
         }
 
         Ok(prices)
     }
+
+    /// Get auction data related to the given settlement.
+    pub async fn get_settlement_auction(
```

Comment:
I think this function is quite long but lends itself very well to be split up into multiple private functions that implement the individual steps.